### DISCUSSION TITLE
Support for non-point queries>

### DISCUSSION THREAD

**Discussion by ItsGarand** [offa/influxdb-cxx#136] [Q&A] [answered]
Is there a way to run a query for non-point data?
like "show databases" or "show measurements"
I would like a way to query influx to check if a database actually exists prior to executing a query for data
Same for measurements

A CURL request to influx returns a JSON stringlike the following:
{"results:[{"statement_id" : 0, "series": [{"name":""databases", "columns": [...]}]}]}...etc.

at a minimum, returning the results string so it could be searched/manipulated outside of the lib would be beneficial

**Comment by offa** [comment, marked as answer]
> Is there a way to run a query for non-point data?

Not yet.

> at a minimum, returning the results string so it could be searched/manipulated outside of the lib would be beneficial

Indeed, some generic "execute something and return the response" sounds useful as a first step.

**Reply by offa** [reply]
#137

**Reply by offa** [reply]
@ItsGarand I have implemented a first iteration (PR #141), could you have a look at it?

**Reply by offa** [reply]
PR has been merged.
